targets/ingester/src/lib/fetchContributions/AnswerExtractor.ts

```diff
@@ -1,117 +1,127 @@
+    }
+    return answer;
   }
 
-  public extractAgreementAnswers(answers: AnswerRaw[]): Answer[] {
-    return this.filterAgreementAnswers(answers)
-      .map((answer) => ({
-        id: answer.id,
-        idcc: answer.agreement.idcc,
-        markdown: answer.markdown,
-        references: answer.references
-          .map(this.createGetRefUrl(answer.agreement.idcc))
-          .sort(this.sortBy("title")),
-      }))
-      .sort(this.sortBy("idcc"));
-  }
-
-  private filterAgreementAnswers(answers: AnswerRaw[]): AgreementAnswerRaw[] {
-    return answers.filter(
-      (answer) =>
-        answer.agreement !== null && this.hasAgreement(answer.agreement.idcc)
-    ) as AgreementAnswerRaw[];
-  }
-
-  private hasAgreement(idcc: string): boolean {
-    const agreement = this.agreements.find(
-      (convention) =>
-        this.comparableIdcc(convention.num) === this.comparableIdcc(idcc)
-    );
-    return agreement ? true : false;
-  }
-
-  private createGetRefUrl(
-    idcc: string
-  ): (ref: ContributionReference) => ContributionReference {
+  private mapKaliRefs = (idcc: string, references: any[]): ExternalRef[] => {
+    if (!references.length) return [];
     const agreement = this.agreements.find(
       (item) => this.comparableIdcc(item.num) === this.comparableIdcc(idcc)
     );
     if (!agreement) {
       throw new Error(`agreement ${idcc} not found `);
     }
-    return function getRefUrl(reference) {
-      switch (reference.category) {
-        case "agreement": {
-          if (reference.dila_id) {
-            reference.url = `https://legifrance.gouv.fr/conv_coll/id/${reference.dila_id}/?idConteneur=${agreement.id}`;
-          } else if (agreement.url) {
-            reference.url = agreement.url;
-          }
-          return reference;
-        }
-        case "labor_code": {
-          if (reference.dila_id) {
-            reference.url = `https://legifrance.gouv.fr/codes/id/${reference.dila_id}`;
-          } else {
-            reference.url =
-              "https://www.legifrance.gouv.fr/codes/id/LEGITEXT000006072050";
-          }
-          return reference;
-        }
-        case null: /* Nothing to do */
-      }
+    return references.map((ref) => {
+      return {
+        title: ref.title,
+        url: `https://legifrance.gouv.fr/conv_coll/id/${ref.kali_article.id}/?idConteneur=${agreement.id}`,
+      };
+    });
+  };
 
-      return reference;
-    };
+  private mapLegiRefs = (references: any[]): ExternalRef[] => {
+    if (!references.length) return [];
+
+    return references.map((ref) => {
+      return {
+        title: ref.legi_article.title,
+        url: "/code-du-travail/" + slugify(ref.legi_article.title),
+      };
+    });
+  };
+
+  private mapOtherRefs = (references: any[]): ExternalRef[] => {
+    if (!references.length) return [];
+
+    return references.map((ref: ExternalRef) => {
+      if (!ref.url) delete ref.url;
+      return ref;
+    });
+  };
```

Comment:
Pas trop fan du delete, car on modifie l'objet en entrée qui peut être utilisé ailleurs. Je proposerai :

```suggestion
    return references.map(({url, ...ref}: ExternalRef) => {
      if (!url) ref;
      return { ...ref, url };
    });
  };
```

Reply:
@m-maillot intéressant comme proposition. Du coup, dans ton cas, le ref ne contient plus url après destructuration ?

Reply:
Ah oui sympa :+1: